contracts/stake/src/stake.rs

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::*;
+
+use canonical_derive::Canon;
+use dusk_bytes::Serializable;
+
+#[derive(Debug, Default, Clone, Copy, Canon, PartialEq, Eq)]
+pub struct Stake {
+    value: u64,
+    eligibility: u64,
+    expiration: u64,
+}
+
+impl Stake {
+    pub const fn new(value: u64, eligibility: u64, expiration: u64) -> Self {
+        Self {
+            value,
+            eligibility,
+            expiration,
+        }
+    }
+
+    pub const fn from_block_height(value: u64, block_height: u64) -> Stake {
+        let epoch = Self::epoch(block_height);
+        let eligibility = block_height + MATURITY + epoch;
+        let expiration = block_height + MATURITY + VALIDITY + epoch;
+
+        Self::new(value, eligibility, expiration)
+    }
+
+    pub const fn epoch(block_height: u64) -> u64 {
+        EPOCH - block_height % EPOCH
+    }
+
+    pub const fn value(&self) -> u64 {
+        self.value
+    }
+
+    pub const fn eligibility(&self) -> u64 {
+        self.eligibility
+    }
+
+    pub const fn expiration(&self) -> u64 {
+        self.expiration
+    }
+
+    pub const fn is_valid(&self, block_height: u64) -> bool {
+        self.eligibility <= block_height && block_height < self.expiration
```

Comment:
there is no more expiration time. A stake is valid until withdrawn. However, for the sake of celerity, we can just deprecate this and port the specifications to the latest version in the near future